cpp/splinepy/utils/nthreads.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <cmath>
+#include <thread>
+
+
+namespace splinepy::utils {
+
+/// N-Thread execution. Queries will be splitted into chunks and each thread
+/// will execute those.
+template <typename Func, typename IndexT>
+void NThreadExecution(
+    const Func& f,
+    const IndexT& total,
+    IndexT nthread /* copy */
+) {
+  // For any negative value, std::thread::hardware_concurrency() will be taken
+  // If you are not satisfied with returned value, use positive value.
+  // For more info:
+  //   https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency
+  // Minimal value that comes out of it will be 0.
+  if (nthread < 0) {
+    nthread = std::thread::hardware_concurrency();
+  }
+
+  // 0 or 1, don't create a thread.
+  if (nthread <= 1) {
+    f(0, total);
+    return;
+  }
+
+  // get chunk size and prepare threads
+  const IndexT chunk_size = std::ceil(total / nthread);
```

Comment:
This will not work, because the division will already be integer operation. I would therefore suggest 
```suggestion
  // Make sure to always round up
  const IndexT chunk_size = (total + 1) / nthread;
```

Reply:
that also doesn't work. I'll fix this